### ISSUE TITLE
"Error: while checking for key's "" existence: appconfiguration.BaseClient#GetKeyValue: Failure responding to request: StatusCode=400" when upgrading from 3.35.0 to 3.36.0

### ISSUE THREAD

**Issue by Etiennepicard** [hashicorp/terraform-provider-azurerm#19748]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.3.6

### AzureRM Provider Version

3.36.0

### Affected Resource(s)/Data Source(s)

azurerm_app_configuration_key

### Terraform Configuration Files

```hcl
resource "azurerm_role_assignment" "app-id-AppConfigurationDataReader-config-appcs" {
  principal_id         = var.app_principal_id
  role_definition_name = "App Configuration Data Reader"
  scope                = var.azure_app_configuration_id
}


resource "azurerm_key_vault_access_policy" "app-keyvault-access" {
  key_vault_id = var.azure_keyvault_id
  tenant_id    = var.tenant_id
  object_id    = var.app_principal_id

  key_permissions = var.have_key_permission_on_keyvault ? [
    "Get",
    "UnwrapKey",
    "WrapKey",
  ] : []

  secret_permissions = [
    "Get"
  ]
}

resource "azurerm_app_configuration_key" "app_kv_configs" {
  for_each = var.kv_configs

  configuration_store_id = var.azure_app_configuration_id

  type         = "kv"
  content_type = ""
  key          = "${var.config_key_prefix}:${each.key}"
  value        = each.value
}

resource "azurerm_app_configuration_key" "app_vault_configs" {
  for_each = var.vault_configs

  configuration_store_id = var.azure_app_configuration_id

  type                = "vault"
  key                 = "${var.config_key_prefix}:${each.key}"
  vault_key_reference = each.value
}
```

### Debug Output/Panic Output

```shell
│ Error: while checking for key's "" existence: appconfiguration.BaseClient#GetKeyValue: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: error response cannot be parsed: {"" '\x00' '\x00'} error: EOF
│
│   with module.app-render-config.azurerm_app_configuration_key.app_vault_configs["CustomerSupportToken:PrimarySecret"],
│   on ..\modules\app_config\config.tf line 35, in resource "azurerm_app_configuration_key" "app_vault_configs":
│   35: resource "azurerm_app_configuration_key" "app_vault_configs" {
│
│ while checking for key's "" existence: appconfiguration.BaseClient#GetKeyValue: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: error response cannot be parsed: {"" '\x00' '\x00'}  
│ error: EOF
```

### Expected Behaviour

Not to crash

### Actual Behaviour

Crash

### Steps to Reproduce

Update the `hashicorp/azurerm` provider from 3.35.0
terraform init -upgrade
terraform plan

**Comment by ziyeqf** [comment]
Hi @Etiennepicard, thanks for reporting.

there is an existing issue(#19711) for it and fixed in #19722. the fix pr has been merged and will be released in v3.37 soon, you can subscribe it to get notification. Please use v3.35 for now.

For any questions please leave comments

**Comment by mybayern1974** [comment]
Close for duplicate
